Add AppImage Packaging to CI

AlleM43 opened the pull request:
AppImage packaging using pkg2appimage to convert the Debian package into an AppImage as part of the packaging workflow.

As mentioned in #252

GyulyVGC commented:
Thank you 🙏 

Will review and test this as soon as I get enough time.

AlleM43 commented:
Alright. I had some issues getting the existing build workflow running without secrets, but the second to last commit in the unsquashed branch (https://github.com/AlleM43/sniffnet/tree/d2577161117d8361c483347430ad3ea1238f3744) is what ended up running the AppImage build to completion. It should work without those modifications as long as the deb builds.

GyulyVGC commented:
I noticed you're packaging it only for `amd64`... I'm wondering how AppImages are usually packed for different arch?
Would it be possible to also support at least `aarch64` or `arm64` or also 32-bit architectures? Would it make sense?

AlleM43 commented:
I haven't really looked into that, partially because the only system I have with a DE installed is amd64, and partially because I thought it best to get amd64 working first and try to expand compatibility later. I can take a look at it though.

GyulyVGC commented:
Thanks. 

Some other observations:
- `sniffnet.yml` should be under the `resources/packaging/linux` folder
- make sure the produced artifact name matches the style of other existing artefacts (maybe something like `Sniffnet_LinuxAppImage_arch.AppImage`)
- if it's possible, execute a test workflow on dispatch from your repo so we can test the produced artifact

AlleM43 commented:
I've moved the yaml recipe and changed the artifact name. 

https://github.com/AlleM43/sniffnet/actions/runs/16672113220 produced AppImages for amd64, i386, arm64, and armhf. I haven't tested running them, but I tested amd64 and i386 appimages built locally with the same recipe, and checked all executables and libraries in each AppDir to ensure they were for the correct platfor

I also changed cargo clean to be a separate step in order to deal with an issue I was having with cross. (https://github.com/cross-rs/cross/issues/724)

GyulyVGC commented:
Oh ok, then I guess I can move it back to the original place.

Again thanks so much for your time 🙌

GyulyVGC commented:
I just had a doubt @AlleM43 

Are we sure libpcap is the only needed dependency? See https://github.com/GyulyVGC/sniffnet/wiki/Required-dependencies

It may work on our systems if we already have the other dependencies installed, but not sure if it works in general...

AlleM43 commented:
The other dependencies are part of those assumed by the pkg2appimage devs to be present on every system (see https://github.com/AppImageCommunity/pkg2appimage/blob/master/excludedeblist) and would thus not be installed anyway.
